lib/metasploit/framework/ntds/account.rb

```diff
@@ -0,0 +1,178 @@
+        attr_accessor :logon_time
+        # @!attribute name
+        #   @return [String] The samAccountName of the account
+        attr_accessor :name
+        # @!attribute no_expire
+        #   @return [TrueClass] If the AD account password does not expire
+        #   @return [FalseClass] If the AD account password does expire
+        attr_accessor :no_expire
+        # @!attribute no_pass
+        #   @return [TrueClass] If the AD account does not require a password
+        #   @return [FalseClass] If the AD account does require a password
+        attr_accessor :no_pass
+        # @!attribute nt_hash
+        #   @return [String] The NT Hash of the current password
+        attr_accessor :nt_hash
+        # @!attribute nt_history
+        #   @return [Array<String>] The NT hashes for previous passwords, up to 24
+        attr_accessor :nt_history
+        # @!attribute nt_history_count
+        #   @return [Fixnum] The count of historical NT hashes
+        attr_accessor :nt_history_count
+        # @!attribute pass_date
+        #   @return [String] Human Readable Date for the last password change
+        attr_accessor :pass_date
+        # @!attribute pass_time
+        #   @return [String] Human Readable Time for the last password change
+        attr_accessor :pass_time
+        # @!attribute rid
+        #   @return [Fixnum] The Relative ID of the account
+        attr_accessor :rid
+        # @!attribute sid
+        #   @return [String] Byte String for the Account's SID
+        attr_accessor :sid
+
+        # @param raw_data [String] the raw 3948 byte string from the wire
+        # @raise [ArgumentErrror] if a 3948 byte string is not supplied
+        def initialize(raw_data)
+          raise ArgumentError, "No Data Supplied" unless raw_data.present?
+          raise ArgumentError, "Invalid Data" unless raw_data.length == 3948
+          data = raw_data.dup
+          @name = get_string(data,40)
+          @description = get_string(data,2048)
+          @rid = get_int(data)
+          @disabled = get_boolean(data)
+          @locked = get_boolean(data)
+          @no_pass = get_boolean(data)
+          @no_expire = get_boolean(data)
+          @expired = get_boolean(data)
+          @logon_count = get_int(data)
+          @nt_history_count = get_int(data)
+          @lm_history_count = get_int(data)
+          @expiry_date = get_string(data,30)
+          @logon_date =  get_string(data,30)
+          @logon_time = get_string(data,30)
+          @pass_date = get_string(data,30)
+          @pass_time = get_string(data,30)
+          @lm_hash = get_string(data,33)
+          @nt_hash = get_string(data,33)
+          @lm_history = get_hash_history(data)
+          @nt_history = get_hash_history(data)
+          @sid = data
+        end
+
+        # @return [String] String representation of the account data
+        def to_s
+          <<EOS
```

Comment:
You don't have to put the heredoc contents all the way to the left to avoid the whitespace prefix, you can do `<<-EOS` (note the `-` after `<<` and then `strip_heredoc` to remove the whitespace prefix:

```ruby
  <<-EOS.strip_heredoc
  #{@name} (#{@description})
  #{ntlm_hash}
  Password Expires: #{@expiry_date}
  Last Password Change: #{@pass_time} #{@pass_date}
  Last Logon: #{@logon_time} #{@logon_date}
  Logon Count: #{@logon_count}
  #{uac_string}
  Hash History:
  #{hash_history}
  EOS
```